meetings/2022-09-28.md

```diff
@@ -0,0 +1,187 @@
+* Matthieu: I think that requiring a client_id claim in ID tokens is a mistake. Clients should be required only if agent is using a Client.
+* ...: That said, maybe you wouldn't use Solid-OIDC for unmitigated resource access.
+* Elf: OK, but no client is a bit unrealistic because whatever performs the HTTP request is the client.
+* Matthieu: I think it's important to make the distinction between User-Agents (say a Browser, CurL, Postman...) and a client in the Solid sense, that is, an HTTP application identifying itself via a Client ID Document.
+* ...: I think in the latter case, Solid will very much care about restricting access based on client identifier documents.
+* ...: But in the former case, there is no natural way for the client to assert its identity. Hence it falls in the "Any" vs "Public" (or no) client category of distinction.
+* ...: A CurL running on your desktop cannot authenticate in the Solid OIDC sense... Hence there would be no client to match in that case. Which is different from matching any client.
+* Elf: https://github.com/solid/solid-oidc/blob/main/alternative-flows.md
+* Elf: What do you use for authentication there?
+* Matthieu: That's the thing. I don't really want to discuss Solid OIDC, but I want to acknowledge the fact that an Authorization Server might receive a request in a client-less context. And specific rules should be possible to express against that kind of scenarios (in other words, being able to say in ACP that it is different to have 1 or no client).
+* Matthieu: An "AuthenticatedClient" matcher will fail if there is not a client. A "PublicClient" matcher will succeed if there is not a client.
+* Elf: So it's like "AuthenticatedAgent", the agent needs to be present if "PublicAgent" it will not need to have an agent.
+* ...: Ok, so ACP uses public agent in the same sort of way as WAC?
+* Matthieu: Yes.
+* Elf: This sounds like [my comment](https://github.com/solid/authorization-panel/issues/292#issuecomment-1063600607) on secure defaults for client access issue.
+* Elf: What happens when a matcher does not specify a client attribute compared to the case where the matcher specifies a client attribute matching the "PublicClient"?
+* Matthieu: If a matcher specifies a single client attribute matching the "PublicClient", then it will be satisfied if any or no client is used (in other words, all contexts).
+* ...: If a matcher does not specify a client attribute, then it will need to match on something else (for example, agent or issuer...). In other words, a Matcher that does not specify a client attribute will not be satisfied by a client but might be satisfied by something else.
+* Matthieu: Via Authenticated client now you could say that a client is required (disallow unmitigated access). Consider the following policy:
+
+```ttl
+[]
+  acp:deny acl:Read, acl:Append, acl:Write, acl:Control ;
+  acp:allOf [ acp:client acp:PublicClient ] ;
+  acp:noneOf [ acp:client acp:AuthenticatedClient ] .
+```
```

Comment:
Every line from 93 to 98 needs to be indented with 2 (additional) space chars, so that block gets rendered under/within Matthieu's `Consider the following policy:` bullet.  I cannot submit this as a suggestion, because GitHub can't handle the 3-backtick code fences within a suggestion.